workflow/load/src/main/java/com/datastax/oss/dsbulk/workflow/load/LoadWorkflow.java

```diff
@@ -351,11 +362,19 @@ public String toString() {
     }
   }
 
+  private int determineWriteConcurrencyWithoutSampling() {
+    return determineWriteConcurrency(() -> 0D);
+  }
+
   private int determineWriteConcurrency() {
+    return determineWriteConcurrency(this::getMeanRowSize);
+  }
+
+  private int determineWriteConcurrency(Supplier<Double> meanRowSizeSupplier) {
     if (dryRun) {
       return numCores;
     }
-    double meanSize = getMeanRowSize();
+    double meanSize = meanRowSizeSupplier.get();
```

Comment:
Wouldn't it be simpler to check support for data size sampling inside `getMeanRowSize` directly?

```
  private double getMeanRowSize() {
    double meanSize;
    try {
      if (!connector.supports(CommonConnectorFeature.DATA_SIZE_SAMPLING)
          || !engineSettings.isDataSizeSamplingEnabled()) {
        return _1_KB;
      }
      LOGGER.debug("Sampling data...");
      ...
  }
```

Reply:
Don't you think that `getMeanRowSize` is already complicated enough? I think that the current design with two methods states clearly the intent.

Reply:
It's the `Supplier<Double>` that bothers me.